tests/clients/test_experiment.py

```diff
@@ -312,11 +337,18 @@ def test_tag_schema_dump():
     assert data == TAG_BODY
 
 
-def test_experiment_run_data_schema():
+@pytest.mark.parametrize(
+    "metric,metric_body",
+    [[METRIC, METRIC_BODY], [METRIC_WITHOUT_STEP, METRIC_WITHOUT_STEP_BODY]],
+)
+def test_experiment_run_data_schema(metric, metric_body):
     data = ExperimentRunDataSchema().dump(
-        {"metrics": [METRIC], "params": [PARAM], "tags": [TAG]}
+        {"metrics": [metric], "params": [PARAM], "tags": [TAG]}
     )
-    assert data == EXPERIMENT_RUN_DATA_BODY
+
+    EXPECTED_EXPERIMENT_RUN_DATA_BODY = EXPERIMENT_RUN_DATA_BODY
+    EXPECTED_EXPERIMENT_RUN_DATA_BODY["metrics"] = [metric_body]
```

Comment:
This code actually modifies the module-level variable. The line above is just making a second reference to the same (mutable) dictionary, which we are then mutating with the second line.

I suggest something like:

```python
    expected = EXPERIMENT_RUN_DATA_BODY.copy()
    expected["metrics"] = [metric_body]
```

Note that I've made the variable non-caps as that's a Python convention meant to indicate that a variable should be treated as a constant (Python does not have actual constants encoded in the language).

Reply:
For the sake of clarity, I think that my suggestion above would not work if mutating some more deeply nested data structure. From what I remember, `.copy()` only makes a 'shallow copy' of the dictionary, which does not actually copy the contents of any collections inside (it makes a reference).

Reply:
Yeah sorry you are right. As we were writing this code I told myself to double check this was fine, because I know it wasn't very idiomatic, and then forgot. Good catch